### ISSUE TITLE
Log file not created during installation (/var/log/neo4j/debug.log)

### ISSUE THREAD

**Issue by remyleone** [neo4j/neo4j#11698]
## Guidelines

- Neo4j version: 3.3.5
- Operating system: (Ubuntu 16.04)

### Steps to reproduce

From Ubuntu base image

```
apt-get update
apt-get install -y wget apt-transport-https
wget -O - https://debian.neo4j.org/neotechnology.gpg.key | apt-key add -
echo 'deb https://debian.neo4j.org/repo stable/' | tee -a /etc/apt/sources.list.d/neo4j.list
apt-get update
apt-get install -y neo4j
```

### Expected behavior

`/var/log/neo4j/debug.log` is created and neo4j can write on it.

### Actual behavior

I get the following error when neo4j starts:

```
May 02 15:13:31 neo4j-genealogy systemd[1]: Started Neo4j Graph Database.
May 02 15:13:31 neo4j-genealogy neo4j[9018]: Active database: graph.db
May 02 15:13:31 neo4j-genealogy neo4j[9018]: Directories in use:
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   home:         /var/lib/neo4j
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   config:       /etc/neo4j
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   logs:         /var/log/neo4j
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   plugins:      /var/lib/neo4j/plugins
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   import:       /var/lib/neo4j/import
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   data:         /var/lib/neo4j/data
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   certificates: /var/lib/neo4j/certificates
May 02 15:13:31 neo4j-genealogy neo4j[9018]:   run:          /var/run/neo4j
May 02 15:13:31 neo4j-genealogy neo4j[9018]: Starting Neo4j.
May 02 15:13:33 neo4j-genealogy neo4j[9018]: 2018-05-02 15:13:33.658+0000 INFO  ======== Neo4j 3.3.5 ========
May 02 15:13:33 neo4j-genealogy neo4j[9018]: 2018-05-02 15:13:33.760+0000 INFO  Starting...
May 02 15:13:33 neo4j-genealogy neo4j[9018]: 2018-05-02 15:13:33.875+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5123a213' was successfully initialized, but faile
May 02 15:13:33 neo4j-genealogy neo4j[9018]: org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5123a213' was successfully initialized, but failed to start. Pleas
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:68)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:220)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:111)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:79)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
May 02 15:13:33 neo4j-genealogy neo4j[9018]: Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@5123a213' was successfully initialized, but failed to start. Please see
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:466)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:107)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:212)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         ... 3 more
May 02 15:13:33 neo4j-genealogy neo4j[9018]: Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: /var/log/neo4j/debug.log (Permission denied)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.kernel.impl.factory.PlatformModule.createLogService(PlatformModule.java:301)
May 02 15:13:33 neo4j-genealogy neo4j[9018]:         at org.neo4j.kernel.impl.factory.PlatformModule.<init>(PlatformModule.java:161)
May 02 15:13:33 neo4j-genealogy systemd[1]: neo4j.service: Main process exited, code=exited, status=1/FAILURE
May 02 15:13:33 neo4j-genealogy systemd[1]: neo4j.service: Unit entered failed state.
May 02 15:13:33 neo4j-genealogy systemd[1]: neo4j.service: Failed with result 'exit-code'.
May 02 15:13:34 neo4j-genealogy systemd[1]: neo4j.service: Service hold-off time over, scheduling restart.
May 02 15:13:34 neo4j-genealogy systemd[1]: Stopped Neo4j Graph Database.
```

**Comment by jennyowen** [comment]
It looks like the `neo4j` service doesn't have permissions to write to `/var/log/neo4j/`.
What command are you using to start the service?

I assume you're running the reproduction steps as sudo user?

**Comment by remyleone** [comment]
I'm running this script as root. The script I gave you can reproduce the error. I didn't use any other command.

**Comment by jennyowen** [comment]
I tried reproducing this using neo4j 3.3.7 and found that I could only reproduce it if I used `sudo neo4j 
start` to start the neo4j service instead of `systemctl`. The service didn't start automatically for me.

I think the problem is caused by the installer not creating all the necessary folders on installation, so when the neo4j service starts it creates those folders. If the service starts with root permissions then the folders are only writable by `root` and not the `neo4j` user creating inconsistencies and this failure.
I fixed the deb and rpm installers so that all the folders are created on installation. This fixed the issue for me when I tried reproducing it again.

Thanks for reporting this!
The fix will be released in versions 3.3.8, 3.4.10 and 3.5.0-beta03.
